packages/resilience/src/circuit-breaker.ts

```diff
@@ -0,0 +1,147 @@
+import CircuitBreaker from 'opossum';
+// eslint-disable-next-line import/named
+import { AxiosError } from 'axios';
+import {
+  Context,
+  HttpMiddlewareContext,
+  Middleware,
+  MiddlewareIn
+} from './middleware';
+
+type ErrorFilter = (err) => boolean;
+type KeyBuilder<ContextT extends Context> = (context: ContextT) => string;
+
+function httpErrorFilter(error: AxiosError): boolean {
+  if (
+    error.response?.status &&
+    error.response.status.toString().startsWith('4')
+  ) {
+    return true;
+  }
+  return false;
+}
+
+function isAxiosError(err: AxiosError | Error): err is AxiosError {
+  return err['isAxiosError'] === true;
+}
+
+function xsuaaErrorFilter(error: AxiosError | Error): boolean {
+  if (isAxiosError(error)) {
+    if (
+      error.response?.status &&
+      error.response.status.toString().startsWith('4')
+    ) {
+      return true;
+    }
+  } else if (error['statuscode'] && error['statuscode'] === 500) {
```

Comment:
[q] I assume this means some setup error right? It is a bit missleading why 500 are not counted. Perhaps add a comment.